libqtile/widget/base.py

```diff
@@ -592,5 +595,39 @@ class MarginMixin(configurable.Configurable):
     margin_y = configurable.ExtraFallback('margin_y', 'margin')
 
 
+class Mirror(_Widget):
+    def __init__(self, reflection):
+        _Widget.__init__(self, reflection.length)
+        reflection.draw = self.hook(reflection.draw)
+        self.reflects = reflection
+        self._length = 0
+
+    @property
+    def length(self):
+        return self.reflects.length
+
+    @length.setter
+    def length(self, value):
+        self._length = value
+
+    def hook(self, draw):
+        def _():
+            draw()
+            self.draw()
+        return _
+
+    def draw(self):
+        if self._length != self.reflects.length:
+            self._length = self.length
```

Comment:
Do we need this _length member at all? Can we just get rid of it entirely, and have the setter body be `pass`?

Reply:
I think so - line 620 detects if the length has changed, so that if it has changed it will re-draw the whole bar. If it hasn't changed length then it will just re-draw itself. Or am I missing something?

Reply:
I guess I'm wondering why does the length ever diverge from the underlying widget?

Reply:
When the original widget changes length, it draws. When it draws, it afterwards run's the mirrow's draw. At this draw execution the lengths mismatch, so the mirror gets the new length from the original and then re-draws the whole bar. Redrawing the whole bar then executes the mirror's draw a second time, but this time the length matches the original's and draws itself. Make sense?

Reply:
Yeah, I'm just wondering if we can skip that and just re-draw our own with the new length from the underlying widget? Or do we need to do the bar.draw() to recalculate all the offsets?

Reply:
My understanding of this came from [here](https://github.com/qtile/qtile/blob/master/libqtile/widget/base.py#L247), which suggests you ought to redraw the whole bar when the length changes. I guess this is to recalculate the offsets like you say.

Reply:
Ah, yeah, makes sense. Can you squash the fixes? This LGTM.